src/ess/polarization/he3.py

```diff
@@ -247,33 +255,71 @@ def expected_transmission(
-                transmission_empty_glass=transmission_empty_glass,
-            )(time=time, wavelength=wavelength, plus_minus='plus')
-        else:
-            opacity = opacity_function(wavelength)
-            polarization = polarization_function(time)
-            return transmission_incoming_unpolarized(
-                transmission_empty_glass=transmission_empty_glass,
-                opacity=opacity,
-                polarization=polarization,
-            )
-
-    if transmission_fraction.coords.is_edges('wavelength'):
-        transmission_fraction = transmission_fraction.copy(deep=False)
-        transmission_fraction.coords['wavelength'] = sc.midpoints(
-            transmission_fraction.coords['wavelength']
+        opacity = opacity_function(wavelength)
+        polarization = polarization_function(time)
+        return transmission_incoming_unpolarized(
+            transmission_empty_glass=transmission_empty_glass,
+            opacity=opacity,
+            polarization=polarization,
         )
 
     popt, _ = sc.curve_fit(
         ['wavelength', 'time'],
         expected_transmission,
-        transmission_fraction,
+        _with_midpoints(transmission_fraction, 'wavelength'),
+        p0={'C': sc.scalar(0.8, unit=''), 'T1': sc.scalar(400000.0, unit='s')},
+    )
+    # TODO Consider including variances from fit
+    polarization_function = He3PolarizationFunction[PolarizingElement](
+        C=sc.values(popt['C']).data, T1=sc.values(popt['T1']).data
+    )
+    return He3TransmissionFunction[PolarizingElement](
+        opacity_function=opacity_function,
+        polarization_function=polarization_function,
+        transmission_empty_glass=transmission_empty_glass,
+    )
+
+
+def get_he3_transmission_incoming_polarized_from_fit_to_direct_beam(
+    transmission_fraction_up: He3AnalyzerTransmissionFraction[Up, Polarized],
+    transmission_fraction_down: He3AnalyzerTransmissionFraction[Down, Polarized],
+    opacity_function: He3OpacityFunction[Analyzer],
+    transmission_empty_glass: He3TransmissionEmptyGlass[Analyzer],
+) -> TransmissionFunction[Analyzer]:
+    """
+    Transmission function for the analyzer, computed with incoming polarized beam.
+
+    This is composed from the opacity-function and the polarization-function.
+    The implementation fits a time- and wavelength-dependent equation and returns
+    the fitted T(t, lambda).
+
+    DB_pol/DB = T_E * cosh(O(lambda)*P(t))*exp(-O(lambda))
+    """
+    # The two transmission fractions do not share a common time coordinate. Therefore,
+    # we cannot concat along a third dimension for the fit, but concat along time,
+    # with an additional coordinate for the incoming polarization.
+    up = transmission_fraction_up.assign_coords(incoming_polarization=sc.scalar(1))
+    down = transmission_fraction_down.assign_coords(incoming_polarization=sc.scalar(-1))
+    transmission_fraction = sc.concat([up, down], 'time')
```

Comment:
Another question: Can it happen that we have only one of the two?

Reply:
See update please!

Reply:
For now I am assuming we have all four channels. It should be easy to adapt this later if we do not, but I wanted to avoid complicating this further. If it is ok with you then we postpone the (ii) case for later (could be soon, if needed).

Reply:
I think that should be following your code without any changes (incoming_polarization for "up" and "down" will be distinguished, but not the analyzer polarization, which we define as "Up").
Then we leave a note below defining the `expected_transmission()` that this only counts for one analyzer state, i.e., "Up", and that this has to be changed in future

Reply:
> I fear we might forget this afterwards, as the combination of UU and DD is dangerous if the transmission is too much different for UU and DD

I don't understand, can you clarify? Why would it be different or dangerous? What makes one better or worse than the other?

Reply:
If UU and DD have different correction factors, i.e., their intensities should be differently corrected already before going through the analyzer (due to the polarizer and flipper - i.e., only important for the incoming-polarized beam), then the transmission through the analyzer might be slightly different for them even if both are the "parallel" state.

Reply:
So this is not something the software can decide/figure out, but the (instrument?) scientist needs to know, somehow?

Reply:
If e.g. a supermirror is used, and the Flipper after the Supermirror polarizer is not perfect, then for the analyzer the incoming beam Up is not the same anymore as incoming Down, and hence T+ (for the "parallel" state, i.e., to get UU or DD) would "look different" for incoming Up and Down.

Similarly, if a 3He cell is used as polarizer, also T+ and T- for this cell can be slightly different and the same as above can arise.

But I would assume here that this can be a general case, and we have to differ it always for an incoming polarized beam.

Reply:
... but if there are two different T+, wouldn't we also need to distinguish those two for performing the polarization correction, instead of using only the T+ obtained from UU?

Reply:
T+ SHOULD be the same for incoming Up and incoming Down. I mean, if the incoming polarized beam (without sample) would be "perfect", then U should be the same as D, which it would be if we would have performed the correction already just for the polarizer/flipper, i.e., for the analyzer they would look the same. Then they have the same T+ of the analyzer, which should be the case. **But** if we do not perform the correction of intensities for the polarizer/flipper alone before we even compute T+ of the analyzer, I think it would look as if T+ would not be the same.

Reply:
I split it off, see update please (or wait for our call).